### ISSUE TITLE
SCIM MongoDb representationAttributes collection

### ISSUE THREAD

**Issue by gabrielemilan** [simpleidserver/SimpleIdServer#188]
Hello, we have an issue on representationAttributes collection. The documents increased a lot, now we have 4mln documents.

I think there is an issue on delete method.

I not sure but I think that the delete can be done by RepresentationId, instead of attributes ids, am I right?

![image](https://user-images.githubusercontent.com/10979448/129707400-d8e87d23-9633-410d-82b3-dce5eeb6dcb3.png)

**Comment by simpleidserver** [comment]
Hello,

We discovered an issue when a group is assigned to a user.
In fact, when the following request is executed several times then the user will contain more than one empty group :

```
{
 "schemas": [ "urn:ietf:params:scim:api:messages:2.0:PatchOp"],
 "Operations": [ 
 	{ "op": "replace", "path": "userName", "value" : "test" }, 
 	{ "op": "add", "path": "groups", "value": [ { "value" : "60b6ffb6-4230-465a-9f80-053b7e45a34b" } ] }
 ]
}
```

We are going to fix this issue.

Thank your for your remark, your solution is working and is cleaner than the current implementation.
We are going to make some changes in order to improve the Delete operation.

Kind regards,

SimpleIdServer

**Comment by simpleidserver** [comment]
Hello,

We discovered an issue in the HTTP PATCH and PUT operations.
Read only parameters (for example : "Users/Groups") can be updated, because of this issue a user can have one or more empty groups.
This issue is fixed in the branch "release/2.0.0".

Thank you for you remark, your solution is working and is cleaner than the current implementation.
The "Delete" operation has been updated with your remark.

Kind regards,

SimpleIdServer
